src/tankoh2/geometry/dome.py

```diff
@@ -712,6 +720,23 @@ def getDomeResizedByThickness(self, thickness):
         """return a dome that has a resized geometry by given thickness"""
         return DomeConicalIsotensoid(self._rCyl + thickness, self._rPolarOpening, self._rSmall + thickness, self._lRad, self._lCone)
 
+    def getDomeResizedByRCyl(self, deltaRCyl):
```

Comment:
these Methods are always the same - if I checked that corrctly. You could create a class AbstractConicalDome and implement the method there. Then every *ConicalDome class inherits the new class.

Return would be:
return self.__class__(rCyl, self._rPolarOpening, rSmall, lRad, lCone)

Reply:
They are **almost** the same. ConicalEllipticalDome has an additional initialization parameter delta1, but I could overwrite AbstractConicalDome in this case.